src/js/playlist/playlist.js

```diff
@@ -12,32 +12,13 @@ const Playlist = function(playlist) {
 // Go through the playlist and choose a single playable type to play; remove sources of a different type
 export function filterPlaylist(playlist, model, feedData) {
     const list = [];
-    const providers = model.getProviders();
-    const preload = model.get('preload');
-    const itemFeedData = Object.assign({}, feedData);
-    delete itemFeedData.playlist;
 
     playlist.forEach(function(item) {
-        item = Object.assign({}, item);
-
-        item.preload = getPreload(item.preload, preload);
-
-        item.allSources = formatSources(item, model);
-
-        item.sources = filterSources(item.allSources, providers);
+        item = normalizePlaylistItem(model, item, feedData);
```

Comment:
We should still copy `itemFeedData` once (here), then pass it in to each item (so we're not making a new copy of each item):

```js
const itemFeedData = Object.assign({}, feedData);	
delete itemFeedData.playlist;	

playlist.forEach(function(item) {
    item = normalizePlaylistItem(model, item, itemFeedData);
```

Reply:
Just an idea for you to thing about - Consider whether map/filter would be an improvement over defining an array and pushing into it when item is defined:

`return playlist.map((item) => normalizePlaylistItem(model, item, itemFeedData)).filter((item) => !!item)`